### PR TITLE
Add custom filters for orders (adminpages)

### PR DESCRIPTION
**Pull request by mircobabini**
### All Submissions:

* [x] Have you followed the [Contributing guideline](CONTRIBUTING.MD)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Added custom filters capability for osers admin page. This is because in many cases we might need to filter the orders list on specific conditions.

### How to test the changes in this Pull Request:

1. Add a WordPress' filter that adds a custom filter to the order admin page
2. Add a WordPress' filter to check if the orders admin page is loaded with that filter and apply a condition

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: adminpages/orders.php [hunk 1]
```diff
@@ -999,6 +1001,12 @@
 						value="only-paid" <?php selected( $filter, 'only-paid' ); ?>><?php _e( 'Only Paid Orders', 'paid-memberships-pro' ); ?></option>
 					<option 
 						value="only-free" <?php selected( $filter, 'only-free' ); ?>><?php _e( 'Only Free Orders', 'paid-memberships-pro' ); ?></option>
+
+					<?php $custom_filters = apply_filters( 'pmpro_admin_orders_filters', array() ); ?>
+					<?php foreach( $custom_filters as $custom_filter ) { ?>
+						<option 
+							value="<?php $custom_filter['name'] ?>" <?php selected( $filter, <?php $custom_filter['name'] ?> ); ?>><?php $custom_filter['title'] ?></option>
```
Comment (ideadude) [inline adminpages/orders.php:1008]: needs esc_attr_e()

Comment (ideadude) [reply adminpages/orders.php:1008]: and esc_html_e() for the title inside the option.

**Comment by ideadude** [comment]
Looks nice. Just a couple tweaks. I will try to get this into the 2.3 release.

Do you have a gist using these filters that could point to in our docs/blog?

**Comment by ideadude** [comment]
I'm going to merge now and fix the echo/escaping there to get this into the RC2.

**Comment by ideadude** [comment]
Hi there. I've merged this into the 2.3 branch. I changed the way the filter worked a bit. See an example here:

https://gist.github.com/ideadude/5d94276e180f1949a2c734f14865f4c6

**Comment by ideadude** [comment]
Can see my tweaks to your code in here https://github.com/strangerstudios/paid-memberships-pro/commit/7f4864edc1121f8156b0c400c27123dd4f50500e

**Comment by mircobabini** [comment]
Well done! Seems great!

**Pull request merged**